.circleci/wait-until-previous-builds-complete.sh

```diff
@@ -0,0 +1,165 @@
+#! /bin/bash
+
+set -e
+set -o pipefail
+set -o functrace
+
+#********************
+# VARIABLES
+# *******************
+api_root="https://circleci.com/api/v2/"
+# polling branch name. Other branches (PRs) are excluded from consideration.
+branch="master"
+# Polling workflow name. Other workflow names (nightly-tests and deploy-staging) are excluded from consideration.
+workflow_name="build-test-deploy"
+project_slug="gh/all-of-us"
+
+# Max check time on a workflow is 45 minutes because e2e tests may take a long time to finish.
+# DISCLAIMER This max time may not be enough if there are 2 or more workflows in waiting before this workflow.
+max_time=$((45 * 60))
+sleep_time=30
+
+#********************
+# FUNCTIONS
+# *******************
+
+# CIRCLECI_TOKEN is a env variable whose value is personal token.
+# See https://circleci.com/docs/2.0/managing-api-tokens/#creating-a-personal-api-token
+check_circleci_token() {
+  if [[ ! $CIRCLECI_TOKEN ]]; then
+    printf '%s\n' "Required env variable \"CIRCLECI_TOKEN\" is not found.\n Create a personal token then create \"CIRCLECI_TOKEN\" env variable?\n" >&2
+    exit 1
+  fi
+}
+
+post() {
+  local url="${api_root}${1}"
+  printf "curl POST ${url}\n" >/dev/tty
+  curl -X POST \
+    -H "Content-Type: application/json" \
+    -H "Circle-Token: ${CIRCLECI_TOKEN}" \
+    -d "$2" \
+    "${url}"
+}
+
+get() {
+  local url="${api_root}${1}"
+  printf "curl GET ${url}\n" >/dev/tty
+  curl -X GET -sS \
+    -H "Content-Type: application/json" \
+    -H "Circle-Token: ${CIRCLECI_TOKEN}" \
+    "${url}"
+}
+
+# Get list of recently built pipelines. Save results to json file.
+pipeline_json="/tmp/master_branch_pipelines.json"
```

Comment:
nit: it would probably be cleaner to avoid a global filepath like this, i.e. probably return the JSON content directly from this function. Note the caller makes assumptions about the file being populated  right after calling, which is not a great separation of concerns